### PR TITLE
kubestate-metrics manifests

### LINKED ISSUE #206: Move Kubestate Metrics to obs installer

### PR DESCRIPTION
**Pull request by liam-j-bennett**
## Description
Add kubestate-metrics manifests

## Related Issue(s)

Fixes #206 

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: installer/pkg/components/kubestate-metrics/deployment.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/gitpod-io/observability/installer/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func objectPointer[T any](o T) *T {
+	return &o
+}
```
Comment (ArthurSens) [inline installer/pkg/components/kubestate-metrics/deployment.go:16]: Any special reason to not use k8s.io/utils/pointer instead?  🤔

File: installer/pkg/components/kubestate-metrics/prometheusrule.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+import (
+	"github.com/gitpod-io/observability/installer/pkg/common"
+	"github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func prometheusRule() []runtime.Object {
```
Comment (ArthurSens) [inline installer/pkg/components/kubestate-metrics/prometheusrule.go:11]: I thought we were going to import the prometheusRules from YAML for now. Did I miss something?

**Review by ArthurSens — APPROVED** [review]
Nice, just some small comments... but overall looks good 🙂

File: installer/pkg/components/kubestate-metrics/deployment.go (hunk 1 above)
Comment (liam-j-bennett) [reply installer/pkg/components/kubestate-metrics/deployment.go:16]: Not really, other than the above being better 😅 Let the compiler do the work :)

File: installer/pkg/components/kubestate-metrics/prometheusrule.go (hunk 2 above)
Comment (liam-j-bennett) [reply installer/pkg/components/kubestate-metrics/prometheusrule.go:11]: You're right, I forgot. Although it wasn't a huge amount of work, if annoying

**Pull request merged**